src/test/java/software/amazon/cloudwatchlogs/emf/logger/MetricsLoggerTest.java

```diff
@@ -349,7 +351,34 @@ void whenPutMetric_withInvalidValue_thenThrowInvalidMetricException(double value
 
     @Test
     void whenPutMetric_withNullUnit_thenThrowInvalidMetricException() {
-        assertThrows(InvalidMetricException.class, () -> logger.putMetric("test", 1, null));
+        assertThrows(InvalidMetricException.class, () -> logger.putMetric("test", 1, (Unit) null));
+    }
+
+    @Test
+    void whenPutMetric_withNullStorageResolution_thenThrowInvalidMetricException() {
+        assertThrows(
+                InvalidMetricException.class,
+                () -> logger.putMetric("test", 1, (StorageResolution) null));
+    }
+
+    @Test
+    void whenPutMetric_withSameMetricDifferentStorageResolutionAfterFlush_thenAllow()
+            throws InvalidMetricException {
+        logger.putMetric("Count", 1);
+        logger.flush();
+        logger.putMetric("Count", 1.0, StorageResolution.HIGH);
+        logger.flush();
+        assertTrue(sink.getLogEvents().get(0).contains("StorageResolution"));
+    }
+
+    @Test
+    void whenPutMetric_withDifferentStorageResolution_thenThrowInvalidMetricException() {
+        assertThrows(
+                InvalidMetricException.class,
+                () -> {
+                    logger.putMetric("test", 1);
```

Comment:
Can we take this out of the lambda? If `logger.putMetric("test", 1);` throws an `InvalidMetricException` for some reason it should fail the test and not succeed it.

Reply:
Updated!